floyd/client/data.py

```diff
@@ -72,10 +74,19 @@ def create(self, data):
             return response.json().get("id")
 
     def get(self, id):
-        response = self.request("GET",
-                                "{}{}".format(self.url, id))
-        data_dict = response.json()
-        return Data.from_dict(data_dict)
+        try:
+            response = self.request("GET",
+                                    "{}{}".format(self.url, id))
+            data_dict = response.json()
+            return Data.from_dict(data_dict)
+        except ClickException as e:
```

Comment:
I used `ClickException` as a catch-all for the different Exceptions in [floyd.exceptions](https://github.com/floydhub/floyd-cli/blob/master/floyd/exceptions.py#L10-L31) that all subclass from it. These are the exceptions that `BaseClient` raises.

So, when an [exception is raised by `BaseClient`](https://github.com/floydhub/floyd-cli/blob/master/floyd/client/base.py#L57-L81), during a call to `request()`, it's either going to be a `ClickException` (from floyd.exceptions), or an [`HTTPError` raised by `request.raise_for_status()](https://github.com/floydhub/floyd-cli/blob/master/floyd/client/base.py#L81).

I think maybe the intention was to have the exceptions in [floyd.exceptions](https://github.com/floydhub/floyd-cli/blob/master/floyd/exceptions.py) sub-class from `FloydException`, but they actually sub from `ClickException`